### PR TITLE
🐛 Reverting #3239 code and fixing it

### PR DESCRIPTION
**Pull request by GitHK**
## What do these changes do?

- Bring back changes from: https://github.com/ITISFoundation/osparc-simcore/pull/3239  which was reverted by https://github.com/ITISFoundation/osparc-simcore/pull/3277
- fixed issue where volume removal services will hang forever if the dy-sidecar failed to start (replaced wrongly used assert)
- includes some suggestions from https://github.com/ITISFoundation/osparc-simcore/pull/3278 

**Review tips:** @sanderegg @pcrespov @mrnicegyu11 @mguidon please check the commit history of this PR for a simpler review experience.

## Related issue/s

- https://github.com/ITISFoundation/osparc-issues/issues/638
- closes https://github.com/ITISFoundation/osparc-simcore/pull/3278

### CODE REVIEW

**Review by pcrespov — APPROVED** [review]
👍

File: services/director-v2/src/simcore_service_director_v2/core/settings.py [hunk 1]
```diff
@@ -361,6 +361,14 @@ class DynamicServicesSchedulerSettings(BaseCustomSettings):
         5.0, description="interval at which the scheduler cycle is repeated"
     )
 
+    DIRECTOR_V2_DYNAMIC_SCHEDULER_PENDING_VOLUME_REMOVAL_INTERVAL_S: PositiveFloat = (
+        Field(
+            30 * MINS,
+            description="interval at which cleaning of unused dy-sidecar "
+            "docker volume removal services is used",
```
Comment (sanderegg) [inline services/director-v2/src/simcore_service_director_v2/core/settings.py:368]: s: ```is used``` -> ```is executed```

**Review by sanderegg — APPROVED** [review]
I do not really know what you changed... so let's see this second round... but wait before merging...

**Comment by elisabettai** [comment]
Hi @GitHK, for your info while testing this on AWS staging, I've observed some volumes staying there forever. 
study_uuid: `a94dc17c-3efc-11ed-96ce-02420a0b1f13`
both these nodes have volumes still there: `cb581308-75f6-497b-9509-4ae923b46e3a`, `3f06afec-3a9a-4057-a6e2-eaf9c3f44f61`

I don't remember anything weird happening with the sidecar.

For the rest, I've verified that:
- New volumes with unique names appear when re-opening the study
- The volumes get deleted when the study is closed (apart from one case mentioned above)
- I got a dy-sidecar going down spontaneously -> the volume remained
- I've restarted a container and verified that dy-sidecar re-attaches to the existing volume

I'm still testing what happens when putting storage down (a first try of stopping the storage container didn't generate any error in the dy-sidecar while closing the study).

**Comment by elisabettai** [comment]
I've also tested that when the sidecar fails to save (I stopped storage until I got an error in the sidecar), the volumes persist and a manual saving of the data can be triggered.

After manual saving, the volumes are still there. I assume this is expected @GitHK, maybe in a future iteration we can improve the cleaning up of volumes.

**Comment by GitHK** [comment]
> I've also tested that when the sidecar fails to save (I stopped storage until I got an error in the sidecar), the volumes persist and a manual saving of the data can be triggered.
> 
> After manual saving, the volumes are still there. I assume this is expected @GitHK, maybe in a future iteration we can improve the cleaning up of volumes.

I think this is most likely the desired effect. For sure it is the safe way to deal with it. If it becomes an issue we need to address it.

**Pull request merged**